feat: 구매자는 REST API를 통해서 경매를 상세 조회 요청을 할 수 있다.

yudonggeun opened the pull request:
## 📄 Summary
### 변경 사항
- [BuyerAuctionInfo isShowStock 필드 제거](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/67545445fb6c3c6ac63adec19cbf2e4b5a944502)
    - 구매자 조회 전용 DTO에서 isShowStock 정책은 노출하지 않도록 하는 것이 좋다고 생각해서 리펙토링을 진행했습니다.
- [구매자 경매 상세 정보 Controller 구현](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/ac1662bc2bae504e9436557b2ae7513a607175dc)

### 테스트
- [구매자 경매 단건 조회 API 테스트 작성](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/2b08a321c36ec11018be838d8bc08a0b059dd5ed)
    - 경매 가격 정책이 절대값 할인 정책인 경우
    - 경매 가격 정책이 퍼센트 할인 정책인 경우
- [BuyerAuctionInfoFixture 생성](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/3b9e4c81ebf928078c25b42e21833d154fe839de)
    - 빌더 패턴을 적용하여 동적으로 원하는 Fixture를 생성할 수 있도록 합니다.
    - 값을 초기화하지 않는다면 기본으로 설정한 테스트 값이 세팅됩니다.

## 🙋🏻 More
- #214 PR 의 작업이 머지가 되어야 합니다.
    - controller, service 등록, DocumentTest 설정 등이 미반영된 상태입니다.
    - 선행 PR이 머지되면 업데이트를 진행한 이후 버그가 있는지 확인할 예정입니다.


close #137

yudonggeun committed:
fix: BuyerAuctionInfo 필드 제거

- 구매자용 경매 조회 DTO에서는 isShowStock 전략을 노출하지 않도록 변경하였습니다.
- stock은 isShowStock 정책에 따라서 노출 여부가 결정되기 때문에 비노출시에는 0보다는 null이 적합하다고 판단이 되어서 타입을 long -> Long으로 변경하였습니다.

yudonggeun committed:
test: BuyerAuctionInfoFixture 생성

빌더를 통해서 테스트 BuyerAuctionInfo 객체를 생성할 수 있습니다.

yudonggeun committed:
feat: 구매자 경매 상세 정보 조회 API 구현

yudonggeun committed:
test: 구매자 경매 단건 조회 API 테스트 작성

- 경매 가격 정책이 절대값 할인 정책인 경우
- 경매 가격 정책이 퍼센트 할인 정책인 경우

minseok-oh commented on src/main/java/com/wootecam/luckyvickyauction/core/auction/controller/BuyerAuctionController.java:
```diff
@@ -26,9 +30,8 @@ public List<BuyerAuctionSimpleInfo> getAuctions(@RequestBody AuctionSearchCondit
     @GetMapping("/{auctionId}")
-    public BuyerAuctionInfo getAuction(@PathVariable Long auctionId) {
-        // TODO: [Task에 맞게 로직 구현할 것!] [writeAt: 2024/08/16/17:40] [writeBy: chhs2131]
-        throw new UnsupportedOperationException();
+    public BuyerAuctionInfo getAuction(@PathVariable("auctionId") Long auctionId) {
+        return auctionService.getBuyerAuction(auctionId);
```
의견 : ResponseEntity로 한번 감싸주는 것도 좋을 것 같아요!

yudonggeun replied:
반영할께요!

minseok-oh commented on src/test/java/com/wootecam/luckyvickyauction/core/auction/controller/BuyerAuctionControllerTest.java:
```diff
@@ -0,0 +1,99 @@
+                    .willReturn(response);
+
+            // when
+            docsGiven
+                    .accept(MediaType.APPLICATION_JSON_VALUE)
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+                    .when()
+                    .get("/auctions/{auctionId}", auctionId.toString())
+                    .then().log().all()
+                    .apply(document("auctions/{auctionId}",
+                            responseFields(
+                                    fieldWithPath("auctionId").description("경매 ID"),
+                                    fieldWithPath("sellerId").description("판매자 ID"),
+                                    fieldWithPath("productName").description("상품 이름"),
+                                    fieldWithPath("originPrice").description("상품 원가"),
+                                    fieldWithPath("currentPrice").description("현재 가격"),
+                                    fieldWithPath("stock").description("현재 재고"),
+                                    fieldWithPath("maximumPurchaseLimitCount").description("최대 구매 수량 제한"),
+                                    fieldWithPath("pricePolicy").description("가격 정책"),
+                                    fieldWithPath("pricePolicy.type").description("가격 정책 타입"),
+                                    fieldWithPath("pricePolicy.variationWidth").description("절대 변동 폭"),
+                                    fieldWithPath("variationDuration").description("가격 변동 주기"),
+                                    fieldWithPath("startedAt").description("경매 시작 시간"),
+                                    fieldWithPath("finishedAt").description("경매 종료 시간")
+                            )
+                    ))
+                    .statusCode(200);
```
고생하셨습니다..!

minseok-oh reviewed (commented):
코드 잘봤습니다! 👍

yudonggeun committed:
feat: BuyerAuctionController#getAuction return type 변경

- https://github.com/woowa-techcamp-2024/Team7-ELEVEN/pull/219#discussion_r1720885142
- 위의 피드백을 반영하여 ResponseEntity를 반환타입으로 사용하도록 변경합니다.

chhs2131 reviewed (dismissed):
좋아요 좋아요 bb

yudonggeun committed:
Merge branch 'refs/heads/dev' into feat/137-issue

yudonggeun committed:
test: DocumentationTest BuyerAuctionController 등록

yudonggeun committed:
test: BuyerAuctionControllerTest identifier 수정

chhs2131 reviewed (approved):
제가 보기에는 아주 좋습니다!!!!  🙌
